### PR TITLE
fix recursive handling of links

### PR DESCRIPTION
**Pull request by gabemontero**
@bparees PTAL

### CODE REVIEW

**Comment by bparees** [comment]
the test job for this repo may need some work due to the repo renaming (sti-base to s2i-base), not sure
[test]

**Comment by bparees** [comment]
you may also need to rebase if you haven't already to make sure we're testing current code.

**Comment by gabemontero** [comment]
@bparees i rebased before submitting the pull

**Comment by gabemontero** [comment]
I think this is the failing error:

```
+ vagrant ssh -c 'docker tag openshift/base-rhel7-candidate openshift/base-rhel7'

Error response from daemon: no such id: openshift/base-rhel7-candidate
```

**Comment by bparees** [comment]
Yeah but I think that is the correct name so i'm not sure why it isn't
there.  Will have to dig into the repo scripts and vagrant plugin.

Ben Parees | OpenShift

**Comment by gabemontero** [comment]
Still digging, but the entry message from build.sh is:
`-> Building openshift/sti-base-rhel7-candidate ...`

So there is an `sti-` prefix that is not present in the `docker tag openshift/base-rhel7-candidate openshift/base-rhel7`

**Comment by gabemontero** [comment]
BASE_IMAGE_NAME="openshift/${BASE_DIR_NAME#s2i-}" appears to strip "s2i-" when I run it and the preceding lines in build.sh in my "openshift/s2i-base" directory, resulting in "openshift/base" as the later `docker tag ..` seems to expect.

This would imply to me that on jenkins, the repo must be getting extracted to the old "openshift/sti-base" directory.

Further up the log, I see this:
`+ vagrant test-origin-image --image base --source openshift/sti-base --ref 78f31e7820e3b0343ff21134d5478f7773c42857`

which corresponds to this in the job definition:

`vagrant test-origin-image --image base --source openshift/$JOB_NAME --ref $ghprbActualCommit`

Where JOB_NAME is `sti-base`.

So we either change the job name to s2i-base, or we change to job config to not use JOB_NAME in the vagrant checkout.  The latter seems less invasive, I'll wait for @bparees to see this and chime in before pulling the trigger.

**Comment by bparees** [comment]
Makefile for s2i-base should be updated to define the builder_image like our other repos do:
https://github.com/sclorg/s2i-ruby-container/blob/master/Makefile#L2

and hack/build.sh should be updated to expect that value first, like we do here:
https://github.com/sclorg/s2i-ruby-container/blob/master/hack/build.sh#L21-L24

you can just do that in this PR to get things rolling.

**Comment by gabemontero** [comment]
Just setting an env var in the Makefile was not proving sufficient for my testing (both local and the PR).  I had to pass the value as a param into build.sh.

**Comment by bparees** [comment]
You need to do this:

https://github.com/sclorg/s2i-ruby-container/blob/master/hack/common.mk#L16

And the base image name value should be just "base".

**Comment by gabemontero** [comment]
latest updates pushed @bparees

File: hack/build.sh [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 #                  image, if the tests pass.
 
-OS=$1
-
 DOCKERFILE_PATH=""
-BASE_DIR_NAME=$(echo $(basename `pwd`) | sed -e 's/-[0-9]*$//g')
-BASE_IMAGE_NAME="openshift/${BASE_DIR_NAME#s2i-}"
+test -z "$BASE_IMAGE_NAME" && {
+    BASE_DIR_NAME=$(echo $(basename `pwd`) | sed -e 's/-[0-9]*$//g')
+    BASE_IMAGE_NAME="openshift/${BASE_DIR_NAME#s2i-}"
```
Comment (bparees) [inline hack/build.sh]: you still don't need openshift here, it's prepended below.  (though this logic shouldn't be executed now, normally, so it's not the current issue)

**Comment by bparees** [comment]
current failure looks like a flake (the base image tester builds every downstream image and tests it as part of testing the change, so it can be finicky)
[test]

File: hack/build.sh (hunk 1 above)
Comment (gabemontero) [inline hack/build.sh]: removed "openshift/" and pushed ... let's see what the next PR test run reveals

**Comment by gabemontero** [comment]
On Tue, Jun 14, 2016 at 3:17 AM, Ben Parees <notifications@github.com>
wrote:

> current failure looks like a flake (the base image tester builds every
> downstream image and tests it as part of testing the change, so it can be
> finicky)
> [test]
>

Yeah I noticed that with a couple of earlier runs that got past the
original problem.

**Comment by bparees** [comment]
lgtm pending a test pass.

**Comment by gabemontero** [comment]
And we have success in https://ci.openshift.redhat.com/jenkins/job/sti-base/161/

```
Sending e-mails to: jhadvig@redhat.com mfojtik@redhat.com

Setting status of 921c12de429051680114a60880dd10d6e123cdfd to SUCCESS with url https://ci.openshift.redhat.com/jenkins/job/sti-base/160/ and message: ' '
Finished: SUCCESS

```

**Comment by gabemontero** [comment]
Oops ... I see it started https://ci.openshift.redhat.com/jenkins/job/sti-base/161/

**Pull request merged**